### PR TITLE
PR for backend part of ERM-685

### PR DESCRIPTION
**Pull request by EthanFreestone**
Creates a tsv parse endpoint, and a KBART import job class. Changes made to import service and ingest service to allow KBART files to utilise these.

### CODE REVIEW

File: service/grails-app/domain/org/olf/general/jobs/KbartImportJob.groovy [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          packageReference: job.packageReference,
+          packageProvider: job.packageProvider
+        ]
+
+        boolean packageInfoValid = true
+        //TODO Actual validation here
+        if (packageInfo.packageName == null ||
+        packageInfo.packageSource == null ||
+        packageInfo.packageReference == null
+      ) {
```
Comment (sosguthorpe) [inline service/grails-app/domain/org/olf/general/jobs/KbartImportJob.groovy]: Indentation looks off here.

File: service/grails-app/services/org/olf/ImportService.groovy [hunk 2]
```diff
@@ -104,4 +114,251 @@ class ImportService implements DataBinder {
...
+    String packageReference
+    String packageProvider
+
+    if (packageInfo.packageName == null ||
+        packageInfo.packageSource == null ||
+        packageInfo.packageReference == null
+      ) {
+        log.error("Import is missing key package information")
+        return packageImported
+      } else {
```
Comment (sosguthorpe) [inline service/grails-app/services/org/olf/ImportService.groovy]: Indentation

File: service/grails-app/services/org/olf/ImportService.groovy [hunk 3]
```diff
@@ -104,4 +114,251 @@ class ImportService implements DataBinder {
...
+        outputDate = LocalDate.parse(date, monthYearFormat);
+        break;
+      default:
+        outputDate = LocalDate.parse(date);
+        break;
+    }
+    return outputDate;
+  }
+
+  private List buildCoverage(String[] lineAsArray, Map acceptedFields) {
```
Comment (sosguthorpe) [inline service/grails-app/services/org/olf/ImportService.groovy]: Given the generic nature of this class, can we rename this method as I think it's KBART specific?

Comment (EthanFreestone) [reply service/grails-app/services/org/olf/ImportService.groovy]: Renamed to buildKBARTCoverage

**Pull request merged**